_posts/2018-07-03-04-26-InOperator.md

```diff
@@ -0,0 +1,28 @@
+---
+title : "4.26 IN Operator"
+layout: post
+category: "4. OData features"
+---
+
+# IN Operator
+
+Starting in WebAPI OData V7.0.0 [[ASP.NET](https://www.nuget.org/packages/Microsoft.AspNet.OData/7.0.0) | [ASP.NET Core](https://www.nuget.org/packages/Microsoft.AspNetCore.OData/)], the IN operator is a [supported feature](http://docs.oasis-open.org/odata/new-in-odata/v4.01/cn01/new-in-odata-v4.01-cn01.html#_Toc485385090 "new feature") that enables a shorthand way of writing multiple EQ expressions joined by OR. For example,
+
+`GET /service/Products?$filter=Name eq 'Milk' or Name eq 'Cheese' or Name eq 'Donut'`
+
+can become
+
+`GET /service/Products?$filter=Name in ('Milk', 'Cheese', 'Donut')`
+
+Of the binary expression invoking IN, the left operand must be a single value and the right operand must be a comma-separated list of primitive values or a single expression that resolves to a collection; the expression returns true if the left operand is a member of the right operand.
+
+# Usage
+
+IN operator is supported only for $filter at the moment and hardcoded collections are supported with parentheses. See examples below.
+
+```
+~/Products?$filter=Name in ('Milk', 'Cheese')
```

Comment:
can we add an example for parameter alias for collection values?

Reply:
:shipit:

Reply:
Parameter alias is actually not supported for collection values in ODL side. I opened an [issue](https://github.com/OData/odata.net/issues/1161) to track this.